library/std/src/sys/windows/mutex.rs

```diff
@@ -37,106 +30,42 @@ pub type MovableMutex = Mutex;
 unsafe impl Send for Mutex {}
 unsafe impl Sync for Mutex {}
 
-struct Inner {
-    remutex: ReentrantMutex,
-    held: Cell<bool>,
-}
-
-#[derive(Clone, Copy)]
-enum Kind {
-    SRWLock,
-    CriticalSection,
-}
-
 #[inline]
 pub unsafe fn raw(m: &Mutex) -> c::PSRWLOCK {
-    debug_assert!(mem::size_of::<c::SRWLOCK>() <= mem::size_of_val(&m.lock));
-    &m.lock as *const _ as *mut _
+    m.srwlock.get()
 }
 
 impl Mutex {
     pub const fn new() -> Mutex {
         Mutex {
-            // This works because SRWLOCK_INIT is 0 (wrapped in a struct), so we are also properly
+            // This works because SRWLOCK_INIT is 0 (wrapped in a struct), so we are properly
             // initializing an SRWLOCK here.
-            lock: AtomicUsize::new(0),
+            srwlock: UnsafeCell::new(c::SRWLOCK_INIT),
         }
     }
     #[inline]
     pub unsafe fn init(&mut self) {}
+
+    #[inline(always)]
```

Comment:
We rarely use `#[inline(always)]` in the standard library implementation. Just `#[inline]` should be enough in almost all cases.

Reply:
Sounds good; done.